File: articles/metrics-objectives-and-goodharts-law/prolog/goodhart_risk.pl
risk_score(ProxyGap, Pressure, Gaming, GuardrailPenalty, Score) :-
    Score is (ProxyGap + Pressure + Gaming + GuardrailPenalty) / 4.

% Query example:
% ?- risk_score(0.38, 0.88, 0.72, 1.0, Score).

File: articles/metrics-objectives-and-goodharts-law/go/goodhart_risk.go
package main

import "fmt"

func main() {
	proxyGap := 0.38
	pressure := 0.88
	gaming := 0.72
	guardrailPenalty := 1.0
	score := (proxyGap + pressure + gaming + guardrailPenalty) / 4.0
	fmt.Printf("goodhart_risk_score=%.4f\n", score)
}
